components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts

```diff
@@ -183,17 +193,26 @@ export class StatsService {
   }
 
   getReports(reportQuery: ReportQuery, listParams: any): Observable<any> {
+      return this.getReportsWithPages(reportQuery, listParams, 1, 10).pipe(
+        map(reportCollection => reportCollection.reports));
+  }
+
+  getReportsWithPages(reportQuery: ReportQuery, listParams: any,
+    page: number, pageSize: number): Observable<ReportCollection> {
     const url = `${CC_API_URL}/reporting/reports`;
     const formatted = this.formatFilters(reportQuery);
-    let body = { filters: formatted };
+    let body = { filters: formatted,
+                 page,
+                 per_page: pageSize
+               };
 
     const {sort, order} = listParams;
     if (sort && order) {
       body = Object.assign(body, {sort, order: order.toUpperCase()});
     }
 
     return this.httpClient.post<any>(url, body).pipe(
-      map(({ reports }) => reports));
+      map(({ reports, total }) => new ReportCollection(reports, total)));
```

Comment:
The API response already returned the total number of reports, but we were ignoring it. This new function allows for page selection and returns the total number of reports for all pages.